extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -0,0 +1,135 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import * as vscode from 'vscode';
+import { DialogBase } from './dialogBase';
+import { INotebookService } from '../services/notebookService';
+import { DeploymentProvider, DialogFieldInfo, FieldType } from '../interfaces';
+
+const localize = nls.loadMessageBundle();
+
+export class DeploymentDialog extends DialogBase {
```

Comment:
Love this class. Long term, let's factor out the support for JSON-backed dialog so it can be used elsewhere, and from any extension. For now it works great to check in here and iterate.